docs/pages/plugins-document-changes.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Document changes
+section: plugins
+chapter: JavaScript API
+permalink: /plugins/document-changes
+
+order: 301
+excerpt: Allow plugins to respond to arbitrary changes in the document
+---
+
+Starting with Sketch 59 we're introducing a way for plugin authors to listen to arbitrary changes in the document.
+
+This new API should provide a more scalable and flexible alternative to plugin [Actions](./actions). It provides a mechanism to listen to three types of fundamental change events in documents - changes, removals and additions.
+
+> 🚧 We're releasing this feature early to gather feedback from the community, so the API may change in future. Let us know what you think, [file an issue](https://github.com/BohemianCoding/SketchAPI/issues) for comments, suggestions and bug reports or send an email to [developer@sketch.com](mailto:developer@sketch.com).
+
+> 🚀 Check the [document-change-sketchplugin](https://github.com/BohemianCoding/document-change-sketchplugin) for an example implementation, or keep reading for a quick introduction.
+
+### Change types
+
+The fundamental document change types are:
+
+- **Property changes** e.g. a layer's x/y position changing
+- **Object addition**, e.g. adding a new layer to the canvas
+- **Object removal**, e.g. deleting a layer
+
+A fourth type of change - an object moving around within the document - can also be trivially derived using the `isMove` flag. See the code examples below for more information.
```

Comment:
I wouldn't describe it as a fourth type, instead explain the concept behind it. Maybe just rephrase as _"an object moving within the layer hierarchy is removed from they previous path and re-added at a new path. Such changes have their `isMove` set to true"_ .